packages/std/src/results/cosmos_msg.rs

```diff
@@ -24,6 +24,7 @@ where
     // to call into more app-specific code (whatever they define)
     Custom(T),
     Staking(StakingMsg),
+    Distribution(DistributionMsg),
```

Comment:
Once upon a time, `StakingMsg` was feature gated under `#[cfg(feature = "staking")]`. I think it should be again, and same with DistributionMsg. This means that every contract that uses them is forced to expose the `requires_staking` export flag and then can fail on upload to chains without PoS (I expect some PoA chains and PoE chains to use CosmWasm without staking and delegation).

Reply:
It was introduced in https://github.com/CosmWasm/cosmwasm/pull/300 and at some point along the way we decided to remove it at least for messages. I don't member why. Maybe it was relates to the exhaustive/non-exhaustive and related trouble on the enum. But since it works well for the stargate flag now, I'll just try to add it again.